server/src/main/scala/org/allenai/scienceparse/FeedbackStore.scala

```diff
@@ -28,12 +30,12 @@ object FeedbackStore extends Logging {
     ConnectionPool.singleton(dbUrl, dbUser, dbPassword)
 
     // upgrade the schema if necessary
-    {
-      val dbConfig: Config = config[Config]("org.allenai.scienceparse.Server.db-as-root")
-      val dbUrl = dbConfig.getString("url")
+    val dbRootConfig: Config = config[Config]("org.allenai.scienceparse.Server.db-as-root")
+    if (dbRootConfig[Boolean]("upgradeSchema")){
```

Comment:
If `upgradeSchema` is disabled, and the schema isn't right, it should complain and stop, not just run anyways.

Reply:
If I'm going add server as a compile dependency in another project and call directly into FeedbackStore, I wasn't sure that client should be running db migrations and be configured with a root db user. If you always have to be the root user anyways, then why do I have to configure two separate db users?

Reply:
I guess then I should finish out the full http-based client experience instead. I'm going to go in that direction after all.

Reply:
Eh, OK. Not sure that was necessary. I just want to throw an exception when the schema is wrong and the code has no way to fix it. That's all I meant.

Reply:
Actually, you don't even need the extra flag then. Just if the schema is already correct, don't even attempt to connect as root. If it is, try to connect as root, and that will fail, as it should.

Reply:
But I guess HTTP is also fine.

Reply:
Yeah going over http is fine, it was trivial to add. Maybe it's better to not introduce it as a compile time dependency elsewhere anyways, actually use the benefits of an API.